torch/distributed/fsdp/flat_param.py

```diff
@@ -1247,10 +1247,21 @@ def _use_sharded_flat_param(self) -> None:
         if self._use_orig_params:
             self._use_sharded_views()
             # For the post-forward reshard, we may try to use sharded gradient
-            # views, but for the post-backward reshard, we delay the call to
-            # after the reduce-scatter
+            # views (or unsharded gradient views if a gradient was accumulated
```

Comment:
just for my understanding, is this because we want the grad views to be unsharded since we were in no_sync, we didn't actually reduce_scatter and reshard?

Reply:
Yup